osu.Game/Overlays/Chat/ChatLine.cs

```diff
@@ -66,6 +68,11 @@ public Message Message
 
         private Container? highlight;
 
+        /// <summary>
+        /// if set, it will override <see cref="APIUser.Colour"/> or <see cref="DrawableChatUsername.default_colours"/>.
+        /// </summary>
+        public Color4? UsernameColour { get; init; }
```

Comment:
I am really quite not happy about how complex all of this logic is. Some overriding done here. Some stuff done locally in `DrawableChatUsername`. It's just really quite smeared across.

There should be one place with all this logic. `DrawableChatUsername` should be a dumb component wrt any of this colouring, it should not be deciding what and how to paint the username. It should have `Colour4 AccentColour` and `bool Inverted`, and `ChatLine` should be the one deciding what the values of that is based on the team or the `APIUser` or the randomness factor or whatever.

Reply:
I understand, and I will modify it according to your comment.